lib/eventsource.js

```diff
@@ -90,6 +90,11 @@ function EventSource (url, eventSourceInitDict) {
     // but for now exists as a backwards-compatibility layer
     options.rejectUnauthorized = !!config.rejectUnauthorized
 
+    // If hostname is specified, merge them into the request options (to support IpV6 with a zone-index)
+    if (config.hostname) {
+      options.hostname = config.hostname
+    }
```

Comment:
According to the [nodejs http request docs](https://nodejs.org/api/http.html#httprequestoptions-callback), hostname is an alias for host. I think it's a good idea to set both host and hostname to the same value since they are aliases. Following that doc: `To support [url.parse()](https://nodejs.org/api/url.html#urlparseurlstring-parsequerystring-slashesdenotehost), hostname will be used if both host and hostname are specified.`

Reply:
We can add host also to the options, if specified. Normally a user uses either host or hostname. url.parse() should work if any of them are set. Please let me know if it's OK for you and when a fix of event source lib could be available, because we're waiting for.

Reply:
@ZmRZjFuDsDiR4qi5801F I'm commenting as another maintainer of this repo, since Yusinto is currently occupied with several projects.

First, I'd like to understand what the level of urgency is for you on this. Are you using this package inside of a LaunchDarkly SDK, so that the lack of IPv6 support is preventing you from using LaunchDarkly? Or are you using the package separately for other purposes? We do support the latter, but the primary reason this fork exists is still for use by the LaunchDarkly SDK, because the upstream repo lacked some rarely-used features that we needed; people using SSE for other purposes would normally use the upstream repo, which is more actively maintained. It was not our intention to keep up with every possible use case in this fork.

For the same reason, any changes to this code require a level of review and testing that might be more than you had assumed, because we need to make sure changes do not have any unanticipated effects on usage in the LaunchDarkly SDK, in every version of Node.js that we support.

That said, we do intend to finish reviewing the PR as soon as we have a chance. The reason you didn't get a response two weeks ago after you pushed a change and made this comment is that you hadn't re-requested review, and a single GitHub comment notification is easier to miss than a review request. But, again, I can't make promises as to what the priority will be for this.

Reply:
Hello again, 

sorry for the delay. We're using the package within a Node-RED Node separately:
https://flows.nodered.org/node/node-red-contrib-ctrlx-automation
https://github.com/boschrexroth/node-red-contrib-ctrlx-automation) 

 I just wan't to kindly ask you again, if there's is any chance to merge this "one-liner" (with none known side-effects), doing the job for IpV6 on our Linux Ubuntu Core Hardware, because this is an urgent issue for our customers using stable IPv6 addresses. The developer of our node module told me he can't switch to another SSE library so easily without having a lot of effort there. 

Hopefully answering your questions properly.